### PR TITLE
wxGUI/gui_core: fix import PostGIS DB raster

### PR DESCRIPTION
**Pull request by tmszi**
**Describe the bug**
Import PostGIS DB raster via wxGUI Import raster data dialog doesn't work.

**To Reproduce**

1. Launch wxGUI
2. Export some raster map into GTiff format e.g. `r.out.gdal input=elevation output=/tmp/elevation.tif` and import with `raster2pgsql -I -C -s 3358 /tmp/elevation.tif public.elevation | psql -d <YOUR_DATABASE> -U <YOUR_DATABASE_USER> -h <YOUR_DATABASE_HOST> -p <YOUR_DATABASE_PORT>` to PostGIS DB
3. Set PostGIS DB connection via `db.login` module
4. From the Data catalog toolbar activate Import raster data tool
5. From the Import raster data dialog Source type choose Database RadioButton widget
6. From the Source input Format ComboBox widget choose PostGIS Raster driver
7.  From the Name ComboBox widget choose your database name
8. List of raster layers ListCtrl widget is empty, no raster layers are loaded.

**Expected behavior**
Loading PostGIS DB raster layer and importing via wxGUI Import raster data dialog  should be work.

**Screenshots**

Current behavior:

wxgui_import_raster_data_postgisdb_db_raster_current

Expected behavior:

wxgui_import_raster_data_postgisdb_db_raster_exp

**System description (please complete the following information):**

- Operating System: all
- GRASS GIS version: all

### CODE REVIEW

File: gui/wxpython/gui_core/gselect.py [hunk 1]
```diff
@@ -2103,8 +2104,35 @@ def GetDsn(self):
                 "PostgreSQL",
                 "PostGIS Raster driver",
             ):
+                ret = RunCommand("db.login", read=True, quiet=True, flags="p")
+                message = _(
+                    "PostgreSQL/PostGIS login was not setted."
```
Comment (petrasovaa) [inline gui/wxpython/gui_core/gselect.py:2109]:
```suggestion
                    "PostgreSQL/PostGIS login was not set."
```

File: gui/wxpython/gui_core/gselect.py [hunk 2]
```diff
@@ -2103,8 +2104,35 @@ def GetDsn(self):
...
+                    " Set it via <db.login> module, please."
+                )
+                if not ret:
+                    GError(parent=self, message=message)
+                    return
 
-                dsn = "PG:dbname=%s" % self.dbWidgets["choice"].GetStringSelection()
+                connection_string = None
+                for conn in ret.split(os.linesep):
```
Comment (petrasovaa) [inline gui/wxpython/gui_core/gselect.py:2117]: Any reason not to use splitlines?

```suggestion
                for conn in ret.splitlines():
```

File: gui/wxpython/gui_core/gselect.py [hunk 3]
```diff
@@ -2195,19 +2223,70 @@ def OnUpdate(self, event):
...
+
+                # v.external returns info for individual bands, however projection is shared by all bands ->
+                # (it is possible to take first line)
+
+                lines = ret.splitlines()
+                projectionMatch = "0"
+                if lines:
+                    bandNumber, bandType, projectionMatch = map(
+                        lambda x: x.strip(), lines[0].split(",")
+                    )
```
Comment (petrasovaa) [inline gui/wxpython/gui_core/gselect.py:2290]: Since this part seems to affect other data sources, is this fixing anything?

**Review by petrasovaa — COMMENTED** [review]
I can't review/test this in detail, so I just left couple notes/questions.

Thanks!

File: gui/wxpython/gui_core/gselect.py [hunk 4]
```diff
@@ -2103,8 +2104,35 @@ def GetDsn(self):
                 "PostgreSQL",
                 "PostGIS Raster driver",
```
Comment (petrasovaa) [inline gui/wxpython/gui_core/gselect.py:2105]: Since this is changing code impacting both options, did you test both of them?

Comment (tmszi) [reply gui/wxpython/gui_core/gselect.py:2105]: Yes of course and it works as expected. PR for import PostGIS geometry data is here #2490.

File: gui/wxpython/gui_core/gselect.py (hunk 2 above)
Comment (tmszi) [reply gui/wxpython/gui_core/gselect.py:2117]: Ok, thanks.

**Review by petrasovaa — APPROVED** [review]

**Pull request merged**